pkg/delete/internal/classic/delete.go

```diff
@@ -21,106 +21,140 @@ import (
-			return err
+func Delete(ctx context.Context, client dtclient.Client, theAPI api.API, ps []pointer.DeletePointer) error {
+	client = newCachedDTClient(client)
+	var err2 error
+
+	for i := range ps {
+		e := convertToEntry(ps[i], theAPI)
+
+		var a *api.API
+		var resolvedID string
+		var err error
+		if a, resolvedID, err = resolveIdentifier(ctx, client, &theAPI, e); err != nil && !is404(err) {
+			log.WithFields(field.Coordinate(e.coordinate)).Error("unable to get resolve config ID: %w")
+			err2 = errors.Join(err2, err)
+			continue
 		}
 
-		delValues, err = filterValuesToDelete(logger, entries, values, theApi.ID)
-
-	} else {
-		// for sub-path APIs, it is a bit more complex. we need to query all entries of each scope defined we can delete it.
+		if e.resolvedID == "" {
+			log.WithFields(field.Coordinate(e.coordinate)).Debug("already gone")
+			continue
+		}
 
-		// map all entries by scope, so we can later filter them by scope
-		scopedMapped := map[string][]pointer.DeletePointer{}
-		for _, entry := range entries {
-			scopedMapped[entry.Scope] = append(scopedMapped[entry.Scope], entry)
+		if err := client.DeleteConfigById(*a, resolvedID); err != nil && !is404(err) {
+			log.WithFields(field.Coordinate(e.coordinate)).Error("failed to delete config: %w")
+			err2 = errors.Join(err2, err)
 		}
+	}
 
-		for scope, scopeEntries := range scopedMapped {
-			a := theApi.Resolve(scope)
+	return err2
+}
 
-			var values []dtclient.Value
-			values, err = client.ListConfigs(ctx, a)
+func convertToEntry(p pointer.DeletePointer, theAPI api.API) *entry {
+	var e *entry
+	e = &entry{
+		identifier: p.Identifier,
+		coordinate: p.AsCoordinate(),
+	}
+	if theAPI.Parent != nil {
+		e.parent = &entry{
+			identifier: p.Scope,
+			coordinate: p.AsCoordinate(),
+		}
+	}
+	return e
+}
 
-			if err != nil {
-				var respErr rest.RespError
-				if errors.As(err, &respErr) && respErr.StatusCode == http.StatusNotFound {
-					logger.Debug("No config of type %q found to delete for scope %q :%s", theApi.ID, scope, respErr.Body)
-					err = nil
-				} else {
-					logger.WithFields(field.Error(err)).Error("Failed to fetch existing configs for api %q (scope: %s): %w", a.ID, scope, err)
-					deleteErrs++
-				}
-				continue
-			}
+func is404(err error) bool {
+	var v rest.RespError
+	if errors.As(err, &v) {
+		return v.StatusCode == http.StatusNotFound
+	}
+	return false
+}
 
-			var vals []deleteValue
-			vals, err = filterValuesToDelete(logger, scopeEntries, values, theApi.ID)
+// resolveIdentifier get the actual ID from DT and update entries with it
+func resolveIdentifier(ctx context.Context, client dtclient.Client, theAPI *api.API, e *entry) (*api.API, string, error) {
+	var err error
 
-			delValues = append(delValues, vals...)
+	if theAPI.Parent != nil {
+		if _, _, err := resolveIdentifier(ctx, client, theAPI.Parent, e.parent); err != nil && !is404(err) {
```

Comment:
considering that this is calling itself, what exactly does it do? why do we resolve if we dont care about the result?

Reply:
Ok, I see it is because `e` is being updated.

Reply:
I think probably the complexity introduced by calling itself is not worth the saving in code and would simply be explicit about handling the two cases: with a parent and without.